src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/CVTextMouseActionData.cs

```diff
@@ -0,0 +1,544 @@
+                                        RGDebug.LogVerbose($"CVTextMouseActionData - RequestCVTextEvaluation - botSegment: {segmentNumber} - Request - onSuccess callback - storingResult");
+
+                                        // pick a random rect from the results, if they didn't want this random, they should have specified within rect
+                                        if (cvTextResults.Count > 0)
+                                        {
+                                            var missingWords = new List<string>();
+                                            var textParts = text.Trim().Split(' ').Select(a => a.Trim()).Where(a => a.Length > 0).Select(a => textCaseRule == TextCaseRule.Ignore ? a.ToLower() : a).ToList();
+
+                                            var partIndexes = new Dictionary<string, List<int>>();
+
+                                            //  TODO: Find the possible bounding boxes that contain all the words... pick the smallest , or the one that is 'within rect'
+                                            // if many matches of the same word then this set gets fairly large, but such is life
+
+                                            // We take a multi pass approach here to make this more understandable for future developers reading it.  originally this was a one pass approach but it was too complex/obscure to be serviceable
+                                            // pass1 - get the indexes of each of our textPart words ( remember we may have the same word twice so we'll need to consider that)
+                                            // pass2 - using those index buckets, build out matching sets that include all text parts
+
+                                            var cvTextResultsCount = cvTextResults.Count;
+                                            var textPartsCount = textParts.Count;
+
+                                            // n^2 looping - but we cover the probably large results list 1 time, and the hopefully short lookup text N times
+                                            for (var k = 0; k < cvTextResultsCount; k++)
+                                            {
+                                                var cvTextResult = cvTextResults[k];
+                                                var cvText = cvTextResult.text.Trim();
+
+                                                if (textCaseRule == TextCaseRule.Ignore)
+                                                {
+                                                    cvText = cvText.ToLower();
+                                                }
+
+                                                // see if we can find a matching textPart for this result.. if so.. add this index to that text part tracker
+                                                var matched = false;
+                                                for (var i = 0; !matched && i <textPartsCount; i++)
```

Comment:
This nesting gets pretty crazy XD after we have multiple API-based segments, maybe we can consolidate some things. I imagine there are quite a few large chunks here that may be good utility functions for the object detection work @batu

Reply:
maybe.. but so far each of these has had bespoke processing in the loops :/  I had also hoped to re-use a lot more but ended up writing each of these  mostly from scratch after getting into it.